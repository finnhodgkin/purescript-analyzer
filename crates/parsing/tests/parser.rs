macro_rules! parser_tests {
    ($($name:ident => $source:expr),+ $(,)?) => {
        $(
            #[test]
            fn $name() {
                let lexed = lexing::lex($source);
                let tokens = lexing::layout(&lexed);
                let (node, errors) = parsing::parse(&lexed, &tokens);
                insta::assert_debug_snapshot!((node, errors));
            }
        )+
    };
}

parser_tests!(
    comment_pre_post => include_str!("parser/CommentPrePost.purs"),
    lexer_error_inserted => include_str!("parser/LexerErrorInserted.purs"),
    module_export => include_str!("parser/ModuleExport.purs"),
    module_export_class => include_str!("parser/ModuleExportClass.purs"),
    module_export_class_error_end => include_str!("parser/ModuleExportClassErrorEnd.purs"),
    module_export_comma_error => include_str!("parser/ModuleExportCommaError.purs"),
    module_export_data => include_str!("parser/ModuleExportData.purs"),
    module_export_data_comma_error => include_str!("parser/ModuleExportDataCommaError.purs"),
    module_export_data_item_after_double_period => include_str!("parser/ModuleExportDataItemAfterDoublePeriod.purs"),
    module_export_data_item_before_double_period => include_str!("parser/ModuleExportDataItemBeforeDoublePeriod.purs"),
    module_export_data_item_invalid => include_str!("parser/ModuleExportDataItemInvalid.purs"),
    module_export_empty => include_str!("parser/ModuleExportEmpty.purs"),
    module_export_invalid => include_str!("parser/ModuleExportInvalid.purs"),
    module_export_type_operator => include_str!("parser/ModuleExportTypeOperator.purs"),
    module_export_type_operator_error => include_str!("parser/ModuleExportTypeOperatorError.purs"),
    module_export_type_operator_error_end => include_str!("parser/ModuleExportTypeOperatorErrorEnd.purs"),
    module_header => include_str!("parser/ModuleHeader.purs"),
    module_header_prefixed => include_str!("parser/ModuleHeaderPrefixed.purs"),
);
