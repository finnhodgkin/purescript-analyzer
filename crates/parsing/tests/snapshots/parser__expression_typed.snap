---
source: crates/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..47
      ModuleHeader@0..30
        MODULE@0..6 "module"
        Comment@6..7
          WHITESPACE@6..7 " "
        ModuleName@7..23
          UPPER@7..22 "ExpressionTyped"
          Comment@22..23
            WHITESPACE@22..23 " "
        ModuleExportList@23..23
        WHERE@23..28 "where"
        Comment@28..30
          WHITESPACE@28..30 "\n\n"
      LAYOUT_START@30..30 ""
      ModuleImports@30..30
      ModuleStatements@30..47
        ValueEquation@30..47
          LOWER@30..35 "typed"
          Comment@35..36
            WHITESPACE@35..36 " "
          EquationBinders@36..36
          EquationUnguarded@36..47
            EQUAL@36..37 "="
            Comment@37..38
              WHITESPACE@37..38 " "
            EquationWhere@38..47
              ExpressionTyped@38..47
                ExpressionInteger@38..40
                  INTEGER@38..39 "1"
                  Comment@39..40
                    WHITESPACE@39..40 " "
                DOUBLE_COLON@40..42 "::"
                Comment@42..43
                  WHITESPACE@42..43 " "
                TypeConstructor@43..47
                  QualifiedName@43..47
                    UPPER@43..46 "Int"
                    Comment@46..47
                      WHITESPACE@46..47 "\n"
      LAYOUT_END@47..47 ""
      END_OF_FILE@47..47 ""
    ,
    [],
)
